### PR TITLE
Add azure_rm_accesstoken_info module

### PR DESCRIPTION
**Pull request by puiterwijk**
This adds a module to retrieve an access token for Azure.

##### SUMMARY
This adds a new module to retrieve an Azure access token.
These tokens might be useful for other tasks, for example as a bearer token when pulling DevOps repositories.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_access_token

### CODE REVIEW

**Comment by Fred-sun** [comment]
@puiterwijk Please keep same with the other module file naming format, rename to "azure_rm_accesstoken_info.py"

File: tests/integration/targets/azure_rm_access_token/aliases [hunk 1]
```diff
@@ -0,0 +1 @@
+cloud/azure
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_access_token/aliases]: Add group type!
```suggestion
cloud/azure
shippable/azure/group15
destructive
```

File: plugins/modules/azure_rm_access_token.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+author:
+    - Patrick Uiterwijk (@puiterwijk)
+'''
+
+EXAMPLES = '''
+- name: Get access token for Microsoft Graph
+  azure.azcollection.azure_rm_access_token:
+    scopes:
+    - https://graph.microsoft.com/.default
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]: Wrong indentation: expected 6 but found 4
```suggestion
      - https://graph.microsoft.com/.default
```

File: tests/integration/targets/azure_rm_access_token/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+- name: Get access token for graphql
+  azure.azcollection.azure_rm_access_token:
+    scopes:
+    - https://graph.microsoft.com/.default
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_access_token/tasks/main.yml]: Wrong indentation: expected 6 but found 4
```suggestion
      - https://graph.microsoft.com/.default
```

File: plugins/modules/azure_rm_access_token.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_access_token
+
+version_added: "1.14.0"
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]:
```suggestion
version_added: "1.19.0"
```

File: plugins/modules/azure_rm_access_token.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+short_description: Get Azure API access token
+
+description:
+    - Get an access token for Azure APIs.
+
+options:
+    scopes:
+        description:
+            - The scopes to request.
+        type: list
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]: Missing elements!

Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]:
```suggestion
        type: list
        elements: str
```

File: plugins/modules/azure_rm_access_token.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            - Additional claims required in the token.
+        type: list
+    token_tenant_id:
+        description:
+            - Tenant to include in the token request.
+        type: str
+    enable_cae:
+        description:
+            - Whether to enable Continuous Access Evaluation (CAE) for the requested token.
+        default: False
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]:
```suggestion
        default: false
```

File: plugins/modules/azure_rm_access_token.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+options:
+    scopes:
+        description:
+            - The scopes to request.
+        type: list
+        required: True
+    claims:
+        description:
+            - Additional claims required in the token.
+        type: list
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]: Missing elements
```suggestion
        type: list
        elements: str
```

File: plugins/modules/azure_rm_access_token.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+class AzureRMAccessToken(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            scopes=dict(type='list', elements='str'),
+            claims=dict(type='list', elements='str'),
+            token_tenant_id=dict(type='str'),
+            enable_cae=dict(type='bool'),
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]: Missing default value, 
```suggestion
            enable_cae=dict(type='bool', default=False),
```

File: plugins/modules/azure_rm_access_token.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    scopes:
+    - https://graph.microsoft.com/.default
+'''
+
+RETURN = '''
+access_token:
+    description:
+        - API access token.
+    returned: always
+    type: str
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]: Missing 'sample'

File: plugins/modules/azure_rm_access_token.py [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+expires_on:
+    description:
+        - Timestamp the token expires on.
+    returned: always
+    type: int
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token.py]: Missing 'sample'

**Comment by puiterwijk** [comment]
@Fred-sun thanks for the review! I think I fixed your comments.

File: plugins/modules/azure_rm_access_token_info.py [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+      - https://graph.microsoft.com/.default
+'''
+
+RETURN = '''
+access_token:
+    description:
+        - API access token.
+    returned: success
+    type: str
+    sample: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token_info.py]:
```suggestion
    line too long (167 > 160 characters)
    sample: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
```

File: plugins/modules/azure_rm_access_token_info.py [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+description:
+    - Get an access token for Azure APIs.
+
+options:
+    scopes:
+        description:
+            - The scopes to request.
+        type: list
+        elements: str
+        required: True
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token_info.py]: Argument 'scopes' in argument_spec is not required, but is documented as being required
```suggestion

```

File: plugins/modules/azure_rm_access_token_info.py [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+        self.token_tenant_id = None
+        self.enable_cae = False
+
+        self.results = dict(changed=False)
+
+        super(AzureRMAccessToken, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                 supports_check_mode=True,
+                                                 supports_tags=False,
+                                                 is_ad_resource=False,
+                                                 required_one_of=[['scopes']])
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_access_token_info.py]: required_one_of.0: length of value must be at least 2 @ data['required_one_of'][0]. Got ['scopes']

**Review by Fred-sun — COMMENTED** [review]
@puiterwijk Please keep same with the other module file naming format, rename to "azure_rm_accesstoken_info.py"

**Comment by puiterwijk** [comment]
@Fred-sun sorry about misreading your suggested name the first time! I fixed that and your other remarks.

**Comment by Fred-sun** [comment]
ready_for_review

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**